src/utils/api.ts

```diff
@@ -0,0 +1,20 @@
+export const fetchAPI = async (endpoint: string, method: string, body?: object) => {
+    const baseURL = process.env.NEXT_PUBLIC_API_URL || 'http://localhost:4000';
```

Comment:
Using `string` allows the function consumer to pass any value of type string. We can force the developer to pass only valid options using this union type. 
```suggestion

type HTTPMethod = 'GET' | 'POST' | 'PUT' | 'DELETE' | 'PATCH' | 'OPTIONS' | 'HEAD';

export const fetchAPI = async (
  endpoint: string,
  method: HTTPMethod,
  body?: object
) => {
```